src/mds/MDCache.cc

```diff
@@ -13058,7 +13065,27 @@ void MDCache::repair_inode_stats_work(MDRequestRef& mdr)
 
   mds->server->respond_to_request(mdr, 0);
 }
-
+void MDCache::rdlock_frags_stats_work(MDRequestRef& mdr)
+{
+  CInode *diri = static_cast<CInode*>(mdr->internal_op_private);
+  dout(10) << __func__ << " " << *diri << dendl;
+  if (!diri->is_auth()) {
+    mds->server->respond_to_request(mdr, -ESTALE);
+    return;
+  }
+  if (!diri->is_dir()) {
+    mds->server->respond_to_request(mdr, -ENOTDIR);
+    return;
+  }
+  
+  MutationImpl::LockOpVec lov;
+  lov.add_rdlock(&diri->dirfragtreelock);
+  lov.add_rdlock(&diri->nestlock);
+  lov.add_rdlock(&diri->filelock);
+  if (!mds->locker->acquire_locks(mdr, lov))
+    return;
+  mds->server->respond_to_request(mdr, 0);
```

Comment:
this does not work as expected. we should acquire locks, execute ValidationContinuation::_dirfrag(), then drop locks.